.github/workflows/unittests.yml

```diff
@@ -39,8 +39,36 @@ jobs:
           python -m pip install --user -e .[extras]
       - name: Test project
         run: |
-          python -m pytest --cov=./ --cov-report=xml --durations=50 tests/
+          python -m pytest --cov=./ --cov-report=xml --device="cpu" --durations=50 tests/
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v1.0.10
         with:
           env_vars: OS,PYTHON
+
+  unittest-gpu:
+    runs-on: ubuntu-latest
+    strategy: 
+      fail-fast: false
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install Linux dependencies
+        run: sudo apt-get install libopenblas-dev
+
+      - name: Configure AWS Credentials
```

Comment:
I find this "Secrets are not passed to workflows that are triggered by a pull request from a fork. [Learn more](https://docs.github.com/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). "